pkgs/development/python-modules/pytest/default.nix

```diff
@@ -29,7 +29,7 @@ buildPythonPackage rec {
 
   # Remove .pytest_cache when using py.test in a Nix build
   setupHook = writeText "pytest-hook" ''
-    postFixupHooks+=(
+    postCheckHooks+=(
```

Comment:
At the moment we do not run this for most pytest. Does this mean we have to add `runHook` to every python package with pytest now?

Reply:
This is a bit boiler code one has to remember.

Reply:
Yes, we would have to, unless we get https://github.com/NixOS/rfcs/pull/32. The other option is to set `PYTEST_ADDOPTS` again with `nocacheprovider`. This will, however, break packages that require it, so in those cases we would have to unset `PYTEST_ADDOPTS`.

Neither option is nice, but I prefer us to do what upstream documents: the cache is created by default.

Reply:
Before https://github.com/NixOS/rfcs/pull/32 the non-boilerplate option to add this to a custom phase instead and put that phase into `$preInstallPhases`.

Reply:
Because the `installPhases` are to early I added a `preDistPhase`.